sklearn/preprocessing/_encoders.py

```diff
@@ -636,7 +754,16 @@ def inverse_transform(self, X):
 
         n_samples, _ = X.shape
         n_features = len(self.categories_)
-        n_transformed_features = sum([len(cats) for cats in self.categories_])
+        if self.drop is None:
+            n_transformed_features = sum(len(cats)
+                                         for cats in self.categories_)
+        else:
+            n_transformed_features = sum(len(cats) - 1
```

Comment:
can just do `sum(len(cats) - (drop_idx is not None) for cats, drop_idx in zip(self.categories_, self.drop_idx_)`